builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1107,6 +1111,25 @@ func resourceVSphereVirtualMachineDelete(d *schema.ResourceData, meta interface{
 		}
 	}
 
+	// Safely eject any disks the user marked as keep_on_remove
+	if vL, ok := d.GetOk("disk"); ok {
+		if diskSet, ok := vL.(*schema.Set); ok {
+
+			for _, value := range diskSet.List() {
+				disk := value.(map[string]interface{})
+
+				if v, ok := disk["keep_on_remove"].(bool); ok && v == true {
```

Comment:
Shouldn't this logic be reversed?  We want to NOT delete any that are keep_on_remove.